lightningd/txfilter.c

```diff
@@ -1,14 +1,51 @@
 #include "txfilter.h"
 
 #include <bitcoin/script.h>
+#include <ccan/build_assert/build_assert.h>
 #include <ccan/crypto/ripemd160/ripemd160.h>
+#include <ccan/mem/mem.h>
+#include <common/memleak.h>
+#include <common/pseudorand.h>
 #include <common/utils.h>
+#include <wallet/wallet.h>
 
 struct txfilter {
 	u8 **scriptpubkeys;
 };
 
+struct outpointfilter_entry {
+	struct bitcoin_txid txid;
+	u32 outnum;
+};
+
+static size_t outpoint_hash(const struct outpointfilter_entry *out)
+{
+	struct siphash24_ctx ctx;
+	siphash24_init(&ctx, siphash_seed());
+	siphash24_update(&ctx, &out->txid, sizeof(out->txid));
+	siphash24_u32(&ctx, out->outnum);
+	return siphash24_done(&ctx);
+}
+
+static bool outpoint_eq(const struct outpointfilter_entry *o1,
+			const struct outpointfilter_entry *o2)
+{
+	return memeq(&o1->txid, sizeof(o1->txid), &o2->txid,
```

Comment:
structeq() is preferred over memeq because it make sure they really are the same type.  And it's safe on bitcoin_txid.